src/main/java/org/jboss/tm/TransactionLocal.java

```diff
@@ -26,296 +26,257 @@
+     * @throws IllegalStateException if an error occures while registering a synchronization
+     *     callback with the transaction
+     */
+    public Object get(Transaction transaction) {
+        if (transaction == null) return initialValue();
+
+        Object value = getValue(transaction);
+
+        // is we didn't get a value initalize this object with initialValue()
+        if (value == null) {
+            // get the initial value
+            value = initialValue();
+
+            // if value is null replace it with the null value standin
+            if (value == null) {
+                value = NULL_VALUE;
+            }
+
+            // store the value
+            try {
+                storeValue(transaction, value);
+            } catch (IllegalStateException e) {
+                // depending on the delegate implementation it may be considered an error to
+                // call storeValue after the tx has ended. Further, the tx ending may have
+                // caused the disposal of a previously stored initial value.
+                // for user convenience we ignore such errors and return the initialvalue here.
+                return initialValue();
+            }
+        }
+
+        // if the value is the null standin return null
+        if (value == NULL_VALUE) {
+            return null;
+        }
+
+        // finall return the value
+        return value;
+    }
+
+    /**
+     * Sets the value of this TransactionLocal variable associtated with the thread context
+     * transaction. This is only used to change the value from the one assigned by the initialValue
+     * method, and many applications will have no need for this functionality.
+     *
+     * @param value the value to be associated with the thread context transactions's
+     *     TransactionLocal
+     */
+    public void set(Object value) {
+        set(getTransaction(), value);
+    }
+
+    /**
+     * Sets the value of this TransactionLocal variable associtated with the specified transaction.
+     * This is only used to change the value from the one assigned by the initialValue method, and
+     * many applications will have no need for this functionality.
+     *
+     * @param transaction the transaction for which the value will be set
+     * @param value the value to be associated with the thread context transactions's
+     *     TransactionLocal
+     */
+    public void set(Transaction transaction, Object value) {
+        if (transaction == null) throw new IllegalStateException("there is no transaction");
+        // If this transaction is unknown, register for synchroniztion callback,
+        // and call initialValue to give subclasses a chance to do some
+        // initialization.
+        if (!containsValue(transaction)) {
+            initialValue();
+        }
+
+        // if value is null replace it with the null value standin
+        if (value == null) {
             value = NULL_VALUE;
-         }
```

Comment:
I am going to assume that the deletions are due to reformatting/moving code around. We'll need to test these changes against narayana and/or WildFly to verify nothing functional has been lost.

Reply:
> I am going to assume that the deletions are due to reformatting/moving code around. We'll need to test these changes against narayana and/or WildFly to verify nothing functional has been lost.

Hi @mmusgrov There is nothing deleted only reformated, its difficult to compare here in git, I can see the correct difference in Intellij IDE.